app/models/command.rb

```diff
@@ -44,4 +46,26 @@ def self.usage_ids
     StageCommand.pluck(:command_id) +
     Project.pluck(:build_command_id)
   end
+
+  # We should not destroy the command if it is associated with any stages.
+  # Also, we should not destroy the command if it is the build commands for multiple projects.
+  #
+  # It isn't possible through the UI to associate a build command with multiple projects.
+  # However, it is possible through the API and the console to do this. So in the case that
+  # this does happen, we should prevent such commands from being destroyed.
+  #
+  # When the build command is associated with only one project, it can be destroyed via this code path:
+  # https://github.com/zendesk/samson/blob/master/app/controllers/build_commands_controller.rb#L13
+  def unused?
+    stages.empty? && projects.count <= 1
```

Comment:
there is a usages method for that ... better to use that then re-implement usage counting here